src/pyre_winning.cc

```diff
@@ -0,0 +1,77 @@
+#include "pyre.h"
+#include "process.h"
+#include "pyre_winning.h"
+
+using cyclus::Material;
+using cyclus::Composition;
+using cyclus::toolkit::ResBuf;
+using cyclus::toolkit::MatVec;
+using cyclus::KeyError;
+using cyclus::ValueError;
+using cyclus::Request;
+using cyclus::CompMap;
+
+namespace recycle {
+
+Winning::Winning() {
+  current(0);
+  Rtime(0);
+  flowrate(0);
+  volume(0);
+}
+
+Winning::Winning(double new_current = 8, 
+                 double new_time = 1, 
+                 double new_flowrate = 3, 
+                 double new_volume = 1
+            ) 
+            {
+  set_coeff();
+  current(new_current);
+  Rtime(new_time);
+  flowrate(new_flowrate);
+  volume(new_volume);
+}
+
+void Winning::set_coeff() {
+  c0 = -0.00685;
+  c1 = 0.20413;
+  c2 = -2.273;
+  c3 = 11.2046;
+  c4 = -19.7493;
+  t0 = 0.2903;
+  t1 = -1.696;
+  r0 = 0.12435;
+  r1 = 0.7985;
+}
+
+double Winning::Efficiency() {
+  return Coulombic(c0,c1,c2,c3,c4) * Temporal(t0,t1) * RateEff(r0,r1);
+}
+
+double Winning::Coulombic(double c0 = -0.00685,
+                          double c1 = 0.20413,
+                          double c2 = -2.273,
+                          double c3 = 11.2046,
+                          double c4 = -19.7493
+) {
+  return c0*pow(current(), 4) + c1*pow(current(), 3) + c2*pow(current(), 2)
+          + c3*current() + c4;
```

Comment:
why this is not using the member variable ?

Reply:
same for all the other methods...

Reply:
and default value should be set in the header, I believe

Reply:
The reason for using a getter function for current/other variables is because we are keeping track of the time series and the function pulls it off the back of the vector - I believe this was a suggestion by Katy

Reply:
I am not sure to understand what you are saying, but my point is:
`c0`,`c1`, `c2`, `c3`,` c4` are member variable of the class.

as you name them the same way (and you give them default values), the member variable are never not use.
The question is then, is the class required ? or do you just need a set of method to compute the values you need...
also the default values should be in the header not in the `cc` file, so people can know them when looking at the header (which is installed, the cc in packed in the lib and are not necessary easily accessible)

Reply:
does it make sense ?

Reply:
I think I get what you mean, the coefficients right now will always be their defaults - there currently is no way to change them in the input so they don't need to be a part of the class. Which the only reason they are in the class is for the diverter implementation as seen in the refiner (currently the only place it is put to use)

Reply:
If I get what you're suggesting it's to have these methods in a different file?

Reply:
it depends on their purpose...

for me there is 2 options there:
- keep the method in the class but use the member variables, (if you don't use the method elsewhere it might be fine)
- have a collection of method describing the physics that you will use, and use them in the class:

something like:
in `physics.h`
```
namespace physics {
double Coulombic(double c0=xx, double c1=yx, ...);
}
```

in `physics.cc`
```
namespace physics {
double Coulombic(double c0, double c1, ...) {
[ ..]

}
}
```

in `pyro_winning.h`
```
double Coulombic();
```
in `pyro_winning.cc`
```
double Winning::Coulombic() {
  return physics::Coulombic(c0, c1, ...);
}

Reply:
the choice between 1 and 2 will depends on how often and in how many place you used those methods, if it is only one place, you probably only need the first solution.

Another the question you need to answer is: would you have to call the Coulombic function of a pyro_wining element using coefficient different than the one stored in the class member...

Reply:
Ahhh okay I do understand, I think that physics approach would work and be clean - it would also remove the need for the current situation in `pyre_refining.cc`:
`double Refine::Thermal() {
  return Thermal(temp(),0);
}

double Refine::Thermal(double tmp,
                       double new_eff
) {
  return t0()*pow(tmp,3) + t1()*pow(tmp,2) + t2()*tmp + t3() - new_eff;
}`

Reply:
The coefficients will differ depending on your input facility. For example the values used now are for kaeri's pride facility whereas this function would differ for INL or ANL's